client/my-sites/site-settings/delete-site/index.jsx

```diff
@@ -43,32 +44,206 @@ class DeleteSite extends Component {
+				)
+			) {
+				return translate( 'Before deleting your site, consider exporting its content as a backup' );
+			}
+
+			return translate( '{{strong}}%(siteDomain)s{{/strong}} will be unavailable in the future.', {
+				components: {
+					strong: <strong />,
+				},
+				args: {
+					siteDomain,
+				},
+			} );
+		};
+
 		return (
-			<Notice status="is-warning" showDismiss={ false }>
-				{ translate( '{{strong}}%(siteDomain)s{{/strong}} will be unavailable in the future.', {
+			<Notice status="is-warning" showDismiss={ false } text={ warningText() }>
+				<NoticeAction onClick={ this._checkSiteLoaded } href={ exportLink }>
+					{ translate( 'Export content' ) }
+				</NoticeAction>
+			</Notice>
+		);
+	}
+
+	renderDeleteSiteCTA() {
+		const { isAtomic, isFreePlan, siteDomain } = this.props;
+		const deleteDisabled =
+			typeof this.state.confirmDomain !== 'string' ||
+			this.state.confirmDomain.replace( /\s/g, '' ) !== siteDomain;
+		const isAtomicRemovalInProgress = isFreePlan && isAtomic;
+
+		let deletionText = translate(
+			'Please type in {{warn}}%(siteAddress)s{{/warn}} in the field below to confirm. ' +
+				'Your site will then be gone forever.',
+			{
+				components: {
+					warn: <span className="delete-site__target-domain" />,
+				},
+				args: {
+					siteAddress: this.props.siteId && this.props.siteDomain,
+				},
+			}
+		);
+
+		if (
+			getLocaleSlug() === 'en' ||
+			getLocaleSlug() === 'en-gb' ||
+			i18n.hasTranslation( 'Before deleting your site, consider exporting its content as a backup' )
+		) {
+			deletionText = translate(
+				'Type {{strong}}%(siteDomain)s{{/strong}} below to confirm you want to delete the site:',
+				{
 					components: {
 						strong: <strong />,
 					},
 					args: {
-						siteDomain,
+						siteDomain: this.props.siteDomain,
 					},
-				} ) }
-			</Notice>
+				}
+			);
+		}
+
+		return (
+			<>
+				<p>{ deletionText }</p>
+				<p className="delete-site__deletion-block">
+					<FormTextInput
+						autoCapitalize="off"
+						className="delete-site__confirm-input"
+						onChange={ this.onConfirmDomainChange }
+						value={ this.state.confirmDomain }
+						aria-required="true"
+						id="confirmDomainChangeInput"
+					/>
+					<Button
+						primary
+						scary
+						busy={ this.state.isDeletingSite }
+						disabled={
+							deleteDisabled ||
+							! this.props.siteId ||
+							! this.props.hasLoadedSitePurchasesFromServer ||
+							isAtomicRemovalInProgress
+						}
+						onClick={ this.handleDeleteSiteClick }
+					>
+						{ translate( 'Delete site' ) }
+					</Button>
+				</p>
+			</>
+		);
+	}
+
+	renderBody() {
+		if (
+			getLocaleSlug() === 'en' ||
+			getLocaleSlug() === 'en-gb' ||
+			i18n.hasTranslation(
+				'Deletion is {{strong}}irreversible and will permanently remove all site content{{/strong}} — posts, pages, media, users, authors, domains, purchased upgrades, and previum themes.'
```

Comment:
Small typo here: 
"previum themes" → "premium themes"

Reply:
Thanks @emilyaudela! I'll fix it in the follow-up task: https://github.com/Automattic/wp-calypso/pull/89363